components/raftstore/src/store/peer_storage.rs

```diff
@@ -942,7 +974,14 @@ where
             }
         }
 
-        self.schedule_applying_snapshot();
+        if !res.for_witness {
+            self.schedule_applying_snapshot();
+        } else {
+            // Bypass apply snapshot process for witness as the snapshot is empty, so mark
+            // status as finished directly here
+            let status = Arc::new(AtomicUsize::new(JOB_STATUS_FINISHED));
```

Comment:
has_ready should also need to be set to true, otherwise, it will wait for next tick to know whether snapshot is finished.

Reply:
check L2878 in `fn handle_raft_ready_append()`, it sends a `SnapshotApplied` msg to trigger next round